### DISCUSSION TITLE
Issues with carousel navigation and connection stability on mobile devices

### DISCUSSION THREAD

**Discussion by Daniel-Fei** [zauberzeug/nicegui#3274] [Q&A]
### Question

I am trying to create a simple display page using NiceGUI, where each card shows detailed information about different people. Users should be able to switch between cards/persons by swiping left or right (or click the navigation arrows). Here is the code I tested:

```python
from nicegui import ui

with ui.row().classes('gap-0  w-full'):
    
    with ui.column().classes('w-[0%]'):
        ui.label('')
        
    with ui.column().classes('w-[100%]'):
        
            with ui.carousel(animated=True, arrows=True, navigation=True).classes('items-center w-full').style('height:95vh').props(r'swipeable control-color=blue items-center'):
                    with ui.carousel_slide(), ui.card().classes('bg-blue-100 w-full h-full items-center'):
                        ui.label('Slide 1')
                    with ui.carousel_slide(), ui.card().classes('bg-blue-100 w-full h-full items-center'):
                        ui.label('Slide 2')
                    with ui.carousel_slide(), ui.card().classes('bg-blue-100 w-full h-full items-center'):
                        ui.label('Slide 3')
                        
    with ui.column().classes('w-[0%]'):
        ui.label('')
            
ui.run()
```

This code works fine on my computer using the Edge browser. I can switch between the cards using the navigation arrows and navigation buttons. However, I mainly need this to work on mobile devices (specifically, I tested it on an iPhone 14 pro using the Edge browser v125.0.2535.96), and I encountered several issues:

1. Clicking the navigation arrows/buttons has no effect; **nothing happens** after clicking.
2. I also tried using the **swipeable** property in the code. But I cannot swipe left or right to switch between cards on the phone either.
3. Strangely, when I access the page on my phone, it initially displays correctly, but after a few seconds (less than 10 seconds, maybe around 5 seconds?), the bottom left corner of the page shows the error message "connection lost (trying to reconnect)" and stays in this state. (At the same time, I am sure the program itself did not crash and the page continued to work fine on the computer). The web page on the phone just **keeps showing this error**. If I refresh the page on the phone, it reloads correctly, but after a few seconds, it shows the same error again.

These issues are quite puzzling to me, and I haven't been able to find a solution. Could you please provide some advice or suggestions? Thank you very much!

**Comment by python-and-novella** [comment]
Is there any errors in dev-tools?
Maybe IOS has compatibility issues with quasar？
Try this plz
https://quasar.dev/vue-components/carousel#vertical

**Reply by Daniel-Fei** [reply]
Hi @python-and-fiction, the quasar examples in the above url you provided work fine on my phone

**Reply by Daniel-Fei** [reply]
The official nicegui example(https://nicegui.io/documentation/carousel#carousel) also works well on my phone

**Comment by falkoschindler** [comment]
Strange, I don't see any problem with Chrome on Android or Safari on iOS 17.
Maybe you can further narrow it down by removing props, classes and styling? Maybe iOS 14 is too old?

**Reply by Daniel-Fei** [reply]
Hi @python-and-fiction @falkoschindler,

Sorry for the confusion. After a thorough investigation today, I found that these issues are not related to NiceGUI. It turned out that the problem was caused by the Shadowrocket app I use on my iPhone. When I turn off this app, everything works fine.

Thank you again for your guidance, suggestions, and patience.
